modules/xfem/src/userobjects/ComboCutUserObject.C

```diff
@@ -0,0 +1,119 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComboCutUserObject.h"
+
+registerMooseObject("XFEMApp", ComboCutUserObject);
+
+InputParameters
+ComboCutUserObject::validParams()
+{
+  InputParameters params = GeometricCutUserObject::validParams();
+  params.addClassDescription("Combine multiple geometric cut userobjects.");
+  params.addRequiredParam<std::vector<UserObjectName>>("cuts",
+                                                       "Vector of cut userobjects to combine");
+  params.addRequiredParam<std::vector<GeometricCutSubdomainID>>(
+      "keys", "Possible combinations of the geometric cut subdomain IDs");
```

Comment:
Sorry, I'm not crazy about `keys` and `vals` either. Maybe `input_cut_ids` and `output_cut_ids`?

Reply:
I want to somehow convey the message that we are doing some sort of look-up here. What are synonyms for "key-value"?